Logger/FileLogger.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.IO;
+using static System.Net.WebRequestMethods;
+
+namespace Logger
+{
+    public class FileLogger : BaseLogger
+    {
+        private string _fileName;
+        public string FileName { get => _fileName; set => _fileName = value ?? throw new ArgumentNullException(value); }
```

Comment:
Adding a public setter for `_fileName` goes against what we want here. We are declaring `_fileName` private with the intention of protecting it from the rest of the world and only allowing it to be set within the current class. This is another perfect opportunity for an **auto implemented property** except this one should be readonly.
```suggestion
        public string FileName { get; }
```